### PR TITLE
Jetpack: Remove canManage and remoteManagementUrl

### PR DESCRIPTION
**Pull request by tyxla**
Currently, we compute the `canManage` prop for every Jetpack site. This prop was necessary back in time to signify that Jetpack Manage is available to enable remote management. This prop was available since Jetpack 3.4. However, since Jetpack 7.3 (released almost a year ago), Manage no longer exists and is considered always on (and not disable-able), making this prop unnecessary.

Jetpack's official support policy is to support the current - 1 version. Any older versions aren't supported. This means we can remove any checks that check for 7.9-only or older versions of Jetpack. This basically means removing all of those `canManage` usages. This PR does that, and also removes all of the related components, selectors and tests that are no longer used. 

This PR is a follow-up to #38585, where we did similar cleanup for no longer supported versions of Jetpack.

#### Changes proposed in this Pull Request

* Remove any `canManage` usage from Themes, Plugins and Jetpack Connect
* Delete obsolete `JetpackManageDisabledMessage`
* Delete obsolete `PluginSiteDisabledManage`
* Delete obsolete `canJetpackSiteManage` and `getJetpackSiteRemoteManagementUrl` selectors.
* Update (and clean up) some tests.

#### Testing instructions

* Do some smoke testing with the affected sections for Jetpack sites and look out for errors:
  * Plugins
  * Themes
  * Jetpack Connect
* Verify tests pass

### CODE REVIEW

**Comment by sgomes** [comment]
@tyxla : Sorry, I didn't realise you had this PR in the works and I just merged #39698 😞 Looks like you'll need to rebase.

**Comment by tyxla** [comment]
No problem @sgomes, I've rebased and fixed any conflicts, so this is ready for a review now 😉 Thanks for letting me know it needed a rebase ⭐️

File: client/state/sites/selectors/get-jetpack-computed-attributes.js [hunk 1]
```diff
@@ -16,7 +15,6 @@ export default function getJetpackComputedAttributes( state, siteId ) {
 	return {
 		canAutoupdateFiles: canJetpackSiteAutoUpdateFiles( state, siteId ),
 		canUpdateFiles: canJetpackSiteUpdateFiles( state, siteId ),
-		canManage: canJetpackSiteManage( state, siteId ),
```
Comment (blowery) [inline client/state/sites/selectors/get-jetpack-computed-attributes.js]: this one may still be used by `client/my-sites/types/post-type-unsupported/index.jsx`, could you take a look?

**Review by blowery — APPROVED** [review]
Overall, looks good. Jetpack sections still work with my Jetpack site. canManage for Jetpack sites seems to have disappeared, minus my one comment.

File: client/state/sites/selectors/get-jetpack-computed-attributes.js (hunk 1 above)
Comment (tyxla) [reply client/state/sites/selectors/get-jetpack-computed-attributes.js]: [That's a false alarm](https://github.com/Automattic/wp-calypso/blob/013d5dde76442d5b798fda96e932ddbca4484134/client/my-sites/types/post-type-unsupported/index.jsx#L71) - it's capability check.

**Comment by tyxla** [comment]
Thank you @blowery! ❤️

**Pull request merged**